### DISCUSSION TITLE
Why are options not displayed and QSelect is "loading"?

### DISCUSSION THREAD

**Discussion by wsw70** [quasarframework/quasar#12925] [General - Components / Directives / etc] [answered]
I am trying to display a [QSelect][1] component that will get its entries from an API (that returns choices dependent of what is typed in the box):

    <template>
      <q-select
        v-model="selectedUsers"
        :options="filteredUsers"
        label="assigned to"
        outlined
        input-debounce="0"
        multiple
        style="width: 250px"
        use-chips
        use-input
        @filter="filterUsers"
        new-value-mode="add-unique"
      />
    </template>
    
    <script lang="ts">
    export default {
      name: 'UserChoice',
    }
    </script>
    
    <script lang="ts" setup>
    import { callApi, User } from 'components/helpers'
    import { ref } from 'vue'
    
    const selectedUsers = ref<Array<User>>([])
    const filteredUsers = ref<Array<User>>([])
    
    const filterUsers = (user: string) => {
      callApi({endpoint: `/user/${user}`, method: 'GET'}).
      then(r => {
        console.log(user, r)
        let users: Array<User> = r.Data as Array<User>
        filteredUsers.value = users.map(x => {return {label: x.UserName, value: x}})
      }).
        catch(() => {{}})
    }
    </script>
    
    <style scoped>
    
    </style>

`callAPI` makes a `fetch` call to the API and returns `json()`.

When typing in the box I see

[![enter image description here][2]][2]

The round spinner is spinning and no dropdown with choices is displayed **despite `filteredUsers` having the right content**: an Array of Objects such as 

    {
      label: "John Gremmet",
      value: { <an object with some information }
    }

(I wrote above something that would be successfully filtered against have `emm` in the API - I am checking the content of `filteredUsers` in the Vue extension of DevTools)

**My question: why isn't QSelect using the `filteredUsers`values and spinning (as if it was "loading" something?)**

  [1]: https://quasar.dev/vue-components/select
  [2]: https://i.stack.imgur.com/qAsSJ.png

**Comment by MilosPaunovic** [comment, marked as answer]
Becuase you haven't told your `QSelect` to update values. Check out the first example in `Filtering and autocomplete` section: [https://quasar.dev/vue-components/select#filtering-and-autocomplete](https://quasar.dev/vue-components/select#filtering-and-autocomplete)

**Reply by wsw70** [reply]
Thanks. Tonight I finally managed to understand how this works (at least I think so), and ended up with

```
const filterUsers = (user: string, update: (args: Promise<void>) => void, abort: () => void) => {
  update(callApi({endpoint: `/user/${user}`, method: 'GET'}).
  then(r => {
    console.log(user, r)
    let users: Array<User> = r.Data as Array<User>
    filteredUsers.value = users.map(x => {return {label: x.UserName, value: x}})
  }).
  catch(() => {{}}))
}
```
